jwst/barshadow/bar_shadow.py

```diff
@@ -47,78 +59,105 @@ def do_correction(input_model, barshadow_model):
     # Create output as a copy of the input science data model
     output_model = input_model.copy()
 
+    # Loop over all the slits in the input model
+    corrections = datamodels.MultiSlitModel()
+    for slit_idx, slitlet in enumerate(output_model.slits):
+        slitlet_number = slitlet.slitlet_id
+        log.info('Working on slitlet %d' % slitlet_number)
+
+        if correction_pars:
+            correction = correction_pars.slits[slit_idx]
+        else:
+            correction = _calc_correction(slitlet, barshadow_model, source_type)
+        corrections.slits.append(correction)
+
+        # Apply the correction by dividing into the science and uncertainty arrays:
+        #     var_poission and var_rnoise are divided by correction**2,
+        #     because they're variance, while err is standard deviation
+        if not inverse:
```

Comment:
double negatives?

Reply:
So, yea. In general an attempt is made to put the "default" choice as the initial selection. Hence this construct, which is carried through all the steps. Always open to better naming suggestion.